core/node/rpc/metrics_interceptor.go

```diff
@@ -13,18 +14,22 @@ type streamIdProvider interface {
 }
 
 type metricsInterceptor struct {
-	rpcDuration       *prometheus.HistogramVec
-	unaryInflight     *prometheus.GaugeVec
-	openClientStreams *prometheus.GaugeVec
-	openServerStreams *prometheus.GaugeVec
+	rpcDuration             *prometheus.HistogramVec
+	unaryInflight           *prometheus.GaugeVec
+	unaryStatusCode         *prometheus.GaugeVec
+	openClientStreams       *prometheus.GaugeVec
+	openServerStreams       *prometheus.GaugeVec
+	serverStreamsStatusCode *prometheus.GaugeVec
 }
 
 func (s *Service) NewMetricsInterceptor() connect.Interceptor {
 	return &metricsInterceptor{
-		rpcDuration:       s.rpcDuration,
-		unaryInflight:     s.metrics.NewGaugeVecEx("grpc_unary_inflight", "gRPC unary calls in flight", "proc"),
-		openClientStreams: s.metrics.NewGaugeVecEx("grpc_open_client_streams", "gRPC open client streams", "proc"),
-		openServerStreams: s.metrics.NewGaugeVecEx("grpc_open_server_streams", "gRPC open server streams", "proc"),
+		rpcDuration:             s.rpcDuration,
```

Comment:
i'm not sure why durations is initialized this way, I think it can be removed from Service and be moved here for consistently. (also it's "method" on duration and "proc" here, lets unify)